apps/src/templates/instructions/TopInstructions.jsx

```diff
@@ -725,9 +739,18 @@ class TopInstructions extends Component {
             )}
             {isCSF &&
               this.props.viewAs === ViewType.Teacher &&
-              this.props.teacherMarkdown &&
-              this.state.tabSelected === TabType.TEACHER_ONLY && (
-                <TeacherOnlyMarkdown ref="teacherOnlyTab" />
+              (this.props.hasContainedLevels ||
+                (this.props.teacherMarkdown &&
```

Comment:
is it possible for a level to have a contained level with a solution, but no teacher markdown? if so, you might need to rework this to allow the solution tab to be shown in this case.